### PR TITLE
Check database collation for case sensitivity

### PR DESCRIPTION
**Pull request by lucyzhang929**
Fixes https://github.com/Azure/azure-functions-sql-extension/issues/133

### CODE REVIEW

File: src/SqlAsyncCollector.cs [hunk 1]
```diff
@@ -394,20 +414,40 @@ AS NewData
...
+                    var cmdCollation = new SqlCommand(GetDatabaseCollationQuery(sqlConnection), sqlConnection);
+                    using SqlDataReader rdr = await cmdCollation.ExecuteReaderAsync();
+                    while (await rdr.ReadAsync())
+                    {
+                        caseSensitive = GetCaseSensitivityFromCollation(rdr[Collation].ToString());
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logger.LogInformation($"Encountered exception while retrieving database collation: {ex}. Case insensitive behavior will be used by default.");
```
Comment (Charles-Gagnon) [inline src/SqlAsyncCollector.cs:436]: This should probably be a warning

File: src/SqlAsyncCollector.cs [hunk 2]
```diff
@@ -220,12 +222,11 @@ private static IEnumerable<string> GetExtraProperties(IDictionary<string, string
         /// <param name="table">Information about the table we will be upserting into</param>
         /// <param name="rows">Rows to be upserted</param>
         /// <returns>T-SQL containing data for merge</returns>
-        private static void GenerateDataQueryForMerge(TableInformation table, IEnumerable<T> rows, out string newDataQuery, out string rowData)
+        private static void GenerateDataQueryForMerge(TableInformation table, IEnumerable<T> rows, StringComparer comparer, out string newDataQuery, out string rowData)
```
Comment (Charles-Gagnon) [inline src/SqlAsyncCollector.cs:225]: The table info already has the comparer so we don't need to pass it in separately right?

**Review by Charles-Gagnon — APPROVED** [review]
Please also update the README : https://github.com/Azure/azure-functions-sql-extension/blob/main/README.md?plain=1#L527

File: src/SqlAsyncCollector.cs (hunk 2 above)
Comment (lucyzhang929) [reply src/SqlAsyncCollector.cs:225]: Ah yes I can use table.Comparer. Thanks!

**Comment by lucyzhang929** [comment]
@Charles-Gagnon Should we leave the case-sensitive collation as a known issue in the README for now since we do not yet know when our next release will be?

**Comment by Charles-Gagnon** [comment]
Good point. Can you make a follow up issue to remove that once the update is released?

**Comment by lucyzhang929** [comment]
Sure thing - created an issue here https://github.com/Azure/azure-functions-sql-extension/issues/192

**Pull request merged**